server/aws-lsp-identity/src/sts/cache/fileSystemStsCache.ts

```diff
@@ -0,0 +1,108 @@
+import { StsCache } from './stsCache'
+import { AwsError, Observability } from '@aws/lsp-core'
+import { AwsErrorCodes, IamCredentials } from '@aws/language-server-runtimes/protocol'
+import path, { join } from 'path'
+import { mkdir, readFile, unlink, writeFile } from 'fs/promises'
+import { getHomeDir } from '@smithy/shared-ini-file-loader'
+import { throwOnInvalidCredentialId } from '../../iam/utils'
+
+export class FileSystemStsCache implements StsCache {
+    constructor(private readonly observability: Observability) {}
+
+    async removeStsCredential(name: string): Promise<void> {
+        throwOnInvalidCredentialId(name)
+
+        await unlink(getStsCredentialFilepath(name)).catch(reason => this.ignoreDoesNotExistOrThrow(reason))
+    }
+
+    async getStsCredential(name: string): Promise<IamCredentials | undefined> {
+        return await getStsCredentialFromFile(name)
+            .then(credential => {
+                if (
+                    credential.accessKeyId &&
+                    credential.secretAccessKey &&
+                    credential.sessionToken &&
+                    credential.expiration
+                ) {
+                    // Ensure Expiration is a Date object
+                    if (typeof credential.expiration === 'string') {
+                        credential = { ...credential, expiration: new Date(credential.expiration) }
+                    }
+                    return credential
+                } else {
+                    return undefined
+                }
+            })
+            .catch(reason => this.ignoreDoesNotExistOrThrow(reason))
+    }
+
+    async setStsCredential(name: string, credential: IamCredentials): Promise<void> {
+        if (
+            !(credential.accessKeyId && credential.secretAccessKey && credential.sessionToken && credential.expiration)
+        ) {
+            this.observability.logging.log('File read from STS cache is not an STS credential.')
+            return
+        }
+
+        await writeStsObjectToFile(name, credential).catch(reason => {
+            throw AwsError.wrap(reason, AwsErrorCodes.E_CANNOT_WRITE_SSO_CACHE)
+        })
+    }
+
+    private ignoreDoesNotExistOrThrow(error: unknown): Promise<undefined> {
```

Comment:
I think this is in the SSO variant of this file as well.  If so, consider putting in a common utils library in aws-lsp-core.